app/controllers/transition_checker_email_subscription_controller.rb

```diff
@@ -2,29 +2,18 @@ class TransitionCheckerEmailSubscriptionController < ApplicationController
   before_action :fetch_govuk_account_session
 
   def show
-    oauth_response = OidcClient.new.has_email_subscription(
-      access_token: @govuk_account_session.access_token,
-      refresh_token: @govuk_account_session.refresh_token,
-    )
-    @govuk_account_session.access_token = oauth_response[:access_token]
-    @govuk_account_session.refresh_token = oauth_response[:refresh_token]
+    has_subscription = @govuk_account_session.has_email_subscription?
 
     render json: {
       govuk_account_session: @govuk_account_session.serialise,
-      has_subscription: oauth_response[:result],
+      has_subscription: has_subscription,
```

Comment:
We can shave off another line here if we remove the local variable and just use `has_subscription: @govuk_account_session.has_email_subscription?,`. Not so much to have one less line, but because we only use it here.

Reply:
Actually we can't, because `has_email_subscription?` might refresh the access token, meaning the result of `@govuk_account_session.serialise` will change (unless we start storing tokens in the database rather than the session).  Do you have any suggestions for how that could be made more explicit?

Reply:
Ah, that's interesting. I suppose that's because of
`@access_token = oauth_response[:access_token]` in `oidc_do`, which depends on OpenID Connect's behaviour. I'm not sure how we can make this more explicit in the code without writing comments 🤔

Reply:
I've changed the code to throw an error if you try to make an OAuth call after serialising the session, which will make improper uses clearly visible at least: 940aaf43f4ec7fdb1fb87eac676d3a6e51e8a538

Reply:
That's a good improvement 👍